resolwe_bio/processes/clustering/hierarchical_clustering.py

```diff
@@ -0,0 +1,899 @@
+                )
+
+            if exp.output.exp_type != inputs.exps[0].output.exp_type:
+                self.warning("All expressions must be of the same Expression type.")
+                self.error(
+                    "Expression {} has {} expression type, "
+                    "while sample {} has {} expression type.".format(
+                        inputs.exps[0].entity.name,
+                        inputs.exps[0].output.exp_type,
+                        exp.entity.name,
+                        exp.output.exp_type,
+                    )
+                )
+
+            if exp.output.feature_type != inputs.exps[0].output.feature_type:
+                self.warning("All expressions must be of the same Feature type.")
+                self.error(
+                    "Expression {} has {} feature type, "
+                    "while sample {} has {} feature type.".format(
+                        inputs.exps[0].entity.name,
+                        inputs.exps[0].output.feature_type,
+                        exp.entity.name,
+                        exp.output.feature_type,
+                    )
+                )
+
+            if inputs.preprocessing.genes:
+                if exp.output.source != inputs.preprocessing.source:
+                    self.warning(
+                        "Selected genes must be annotated by the same genome database as all expression files."
+                    )
+                    self.error(
+                        "Gene IDs are from {} database, "
+                        "while sample {} has gene IDs from {} database.".format(
+                            inputs.preprocessing.source,
+                            exp.entity.name,
+                            exp.output.source,
+                        )
+                    )
+                if exp.output.species != inputs.preprocessing.species:
+                    self.warning(
+                        "Selected genes must be from the same species as all expression files."
+                    )
+                    self.error(
+                        "Selected genes are {}, while expression {} is {}.".format(
+                            inputs.preprocessing.species,
+                            exp.entity.name,
+                            exp.output.species,
+                        )
+                    )
+
+            sample_files.append(exp.output.exp.path)
+            sample_ids.append(exp.entity.id)
+            sample_names.append(exp.entity.name)
+
+        if inputs.preprocessing.genes:
+            gene_labels = [gene for gene in inputs.preprocessing.genes]
+
+        if len(gene_labels) == 1 and inputs.processing.distance_metric != "euclidean":
+            self.error(
+                "Select at least two genes to compute hierarchical clustering of samples with "
+                "correlation distance metric or use Euclidean distance metric."
+            )
+
+        if len(sample_files) != len(sample_ids):
```

Comment:
The two checks mentioned here and the corresponding check in hierarchical clustering of genes are now redundant and can be removed. They were necessary before, because the lists were provided as command-line arguments to the python script that computed clustering and we couldn't guarantee consistency otherwise. Now those lists are populated in the `for` loop above which guarantees that all lists contain the same number of items.